x/finality/keeper/gov.go

```diff
@@ -95,9 +103,15 @@ func (k Keeper) HandleResumeFinalityProposal(ctx sdk.Context, fpPksHex []string,
 	}
 
 	// set the all the given finality providers voting power to 0
+	var distCache *ftypes.VotingPowerDistCache
+	allActiveFps := make(map[string]*ftypes.FinalityProviderDistInfo)
 	for h := uint64(haltingHeight); h <= uint64(currentHeight); h++ {
-		distCache := k.GetVotingPowerDistCache(ctx, h)
+		distCache = k.GetVotingPowerDistCache(ctx, h)
 		activeFps := distCache.GetActiveFinalityProviderSet()
+		// add active fps to the all active fps set if it does not exist
+		for pk, fp := range activeFps {
+			allActiveFps[pk] = fp
+		}
```

Comment:
:thinking: This `activeFps` is not being used
```suggestion

```

Reply:
so we don't get the active fps from the latest `*ftypes.VotingPowerDistCache`, but should iterate over the `activeFps` map to verify the signing info

Reply:
Oops. The change is not pushed correctly